.github/workflows/CI.yml

```diff
@@ -207,9 +206,9 @@ jobs:
       VERSION: ${{ needs.prepare_ci_run.outputs.VERSION }}
     steps:
       - name: Install Go
-        uses: actions/setup-go@v3.0.0
+        uses: actions/setup-go@v3.1.0
         with:
-          go-version: 1.17
+          go-version-file: "go.mod"
 
       - name: Checkout Code
         uses: actions/checkout@v3
```

Comment:
You need to move the checkout action before the install go action, in order to be able to read go.mod

Reply:
Updated.